tests/integration/commands/functions-serve/functions-serve.test.mjs

```diff
@@ -0,0 +1,156 @@
+    })
+
+    ps.stdout.on('data', (data) => console.log(data.toString()))
+    ps.stderr.on('data', (data) => console.log(data.toString()))
+
+    const { open } = await waitPort({
+      port,
+      output: 'silent',
+      timeout: SERVE_TIMEOUT,
+    })
+    if (!open) {
+      throw new Error('Timed out waiting for functions server')
+    }
+    return await testHandler()
+  } finally {
+    await killProcess(ps)
+  }
+}
+
+describe.concurrent('functions:serve command', () => {
+  test('should serve functions on default port', async (t) => {
+    await withSiteBuilder('site-with-ping-function', async (builder) => {
+      await builder
+        .withNetlifyToml({ config: { functions: { directory: 'functions' } } })
+        .withFunction({
+          path: 'ping.js',
+          handler: async () => ({
+            statusCode: 200,
+            body: 'ping',
+          }),
+        })
+        .buildAsync()
+
+      await withFunctionsServer({ builder }, async () => {
+        const response = await fetch(`http://localhost:9999/.netlify/functions/ping`)
+        t.expect(await response.text()).toEqual('ping')
+      })
+    })
+  })
+
+  test('should serve functions on custom port', async (t) => {
+    await withSiteBuilder('site-with-ping-function', async (builder) => {
+      await builder
+        .withNetlifyToml({ config: { functions: { directory: 'functions' } } })
+        .withFunction({
+          path: 'ping.js',
+          handler: async () => ({
+            statusCode: 200,
+            body: 'ping',
+          }),
+        })
+        .buildAsync()
+
+      const port = await getPort()
+      await withFunctionsServer({ builder, args: ['--port', port], port }, async () => {
+        const response = await fetch(`http://localhost:${port}/.netlify/functions/ping`)
+        t.expect(await response.text()).toEqual('ping')
+      })
+    })
+  })
+
+  test('should use settings from netlify.toml dev', async (t) => {
+    await withSiteBuilder('site-with-ping-function', async (builder) => {
+      const port = await getPort()
+      await builder
+        .withNetlifyToml({
+          config: { functions: { directory: 'functions' }, dev: { functions: 'other', functionsPort: port } },
+        })
+        .withFunction({
+          pathPrefix: 'other',
+          path: 'ping.js',
+          handler: async () => ({
+            statusCode: 200,
+            body: 'ping',
+          }),
+        })
+        .buildAsync()
+
+      await withFunctionsServer({ builder, port }, async () => {
+        const response = await fetch(`http://localhost:${port}/.netlify/functions/ping`)
+        t.expect(await response.text()).toEqual('ping')
+      })
+    })
+  })
+
+  test('should inject env variables', async (t) => {
+    await withSiteBuilder('site-with-env-function', async (builder) => {
+      await builder
+        .withNetlifyToml({
+          config: {
+            build: { environment: { VITEST_TEST: 'FROM_CONFIG_FILE' } },
```

Comment:
NIT: As mentioned in the other PR I would not rename the variable to something that could be set by vitest (to avoid having it set from the testrunner)

I would rename it to something like `MY_CONFIG`

Reply:
Done!